vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -945,9 +947,13 @@ check_init_state_file(void)
             log_err("Error reading data from \"%s\" in %s",
                     global_domain.info.lat_var, filenames.init_state);
         }
-        for (i = 0; i < global_domain.n_ny; i++) {
-            if (!assert_close_double(dvar[i],
-                                     global_domain.locations[i].latitude, rtol,
+        // implicitly nested loop over ni and nj with i set to 0;
+        // j stride = n_nx
+        for (j = 0; j < global_domain.n_ny; j++) {
+            if (!assert_close_double(dvar[j],
+                                     global_domain.locations[j *
+                                                             global_domain.n_nx]
+                                     .latitude, rtol,
```

Comment:
So this just checks the latitudes in the furthest left (west) grid cells?  Why not loop over `global_domain.ncells` and use the `io_index` to check all coordinate locations?